db/migrate/20210310195204_add_unique_index_to_articles.rb

```diff
@@ -0,0 +1,5 @@
+class AddUniqueIndexToArticles < ActiveRecord::Migration[6.0]
+  def change
+    add_index :articles, [:mw_page_id, :wiki_id], unique: true
```

Comment:
It looks like we have many violations of this constraint in Wiki Ed production:
```
Article.all.group(:mw_page_id, :wiki_id).having('count(*) > 1').count.count
 => 3674 
```

I believe this largely or exclusively represent cases where all but one of the duplicate records are marked as `deleted: true`. This is part of what `ArticleStatusManager` does.

Reply:
This isn't the most critical part of the PR, so I'll think about what I want to do here.

Reply:
Okay. This constraint makes a lot of sense, it's just related to some of the harder parts of the system to reason about. It's been a little while since I've taken a shot at it; we might be able to find a way to make it work.

Reply:
Also, from global production:
```
Article.all.group(:mw_page_id, :wiki_id).having('count(*) > 1').cou
nt.count
 => 16655 
```

Reply:
So, if this is true, how did the code work before? 

Before it was calling 

```ruby 
article = Article.find_by(mw_page_id: article_data['article']['mw_page_id'], wiki_id: @wiki.id)
```

The generated query is:

```
SELECT `articles`.* FROM `articles` WHERE `articles`.`mw_page_id` = 1 AND `articles`.`wiki_id` = 1 LIMIT 1
```

That means that we were selecting a random article out of the set matching these attributes.